pkg/kv/kvserver/rangefeed/catchup_scan.go

```diff
@@ -94,6 +104,41 @@ func (i *CatchUpIterator) CatchUpScan(
 	withDiff bool,
 	outputFn outputEventFn,
 ) error {
+	// We emit any MVCC range tombstones first, so that the caller can use them to
+	// potentially discard point keys below them.
+	//
+	// Note that MVCC range tombstones are currently experimental, and will only
+	// be used when storage.CanUseExperimentalMVCCRangeTombstones() is enabled.
+	if i.rangeKeyIter != nil { // can be nil in tests
+		for {
+			if ok, err := i.rangeKeyIter.Valid(); err != nil {
+				return err
+			} else if !ok {
+				break
+			}
+
+			rangeKey, value := i.rangeKeyIter.Key(), i.rangeKeyIter.Value()
+			if len(value) > 0 {
+				// We currently expect all range keys to be tombstones.
+				panic(errors.AssertionFailedf("unexpected non-tombstone range key %s with value %v",
+					rangeKey, value))
+			}
+			if rangeKey.Timestamp.Less(catchUpTimestamp) {
+				continue
+			}
+			err := outputFn(&roachpb.RangeFeedEvent{
+				DelRange: &roachpb.RangeFeedDeleteRange{
+					Span:      roachpb.Span{Key: rangeKey.StartKey, EndKey: rangeKey.EndKey},
+					Timestamp: rangeKey.Timestamp,
+				},
+			})
+			if err != nil {
+				return err
+			}
+			i.rangeKeyIter.Next()
+		}
+	}
```

Comment:
Do you think it would make sense to move this code into separate helper?  This function is pretty large already, imo.

Reply:
Yeah, good idea. Done.